Assets/MixedRealityToolkit.Services/InputAnimation/InputAnimationSerializationUtils.cs

```diff
@@ -0,0 +1,184 @@
+            for (int i = 0; i < curve.length; ++i)
+            {
+                var keyframe = curve.keys[i];
+                writer.Write(keyframe.time - startTime);
+                writer.Write(keyframe.value);
+                writer.Write(keyframe.inTangent);
+                writer.Write(keyframe.outTangent);
+                writer.Write(keyframe.inWeight);
+                writer.Write(keyframe.outWeight);
+                writer.Write((int)keyframe.weightedMode);
+            }
+        }
+
+        /// <summary>
+        /// Deserialize an animation curve with tangents from binary data.
+        /// </summary>
+        public static void ReadFloatCurve(BinaryReader reader, AnimationCurve curve)
+        {
+            curve.preWrapMode = (WrapMode)reader.ReadInt32();
+            curve.postWrapMode = (WrapMode)reader.ReadInt32();
+
+            int keyframeCount = reader.ReadInt32();
+
+            Keyframe[] keys = new Keyframe[keyframeCount];
+            for (int i = 0; i < keyframeCount; ++i)
+            {
+                keys[i].time = reader.ReadSingle();
+                keys[i].value = reader.ReadSingle();
+                keys[i].inTangent = reader.ReadSingle();
+                keys[i].outTangent = reader.ReadSingle();
+                keys[i].inWeight = reader.ReadSingle();
+                keys[i].outWeight = reader.ReadSingle();
+                keys[i].weightedMode = (WeightedMode)reader.ReadInt32();
+            }
+
+            curve.keys = keys;
+        }
+
+        /// <summary>
+        /// Serialize an animation curve as binary data, ignoring tangents.
+        /// </summary>
+        public static void WriteBoolCurve(BinaryWriter writer, AnimationCurve curve, float startTime)
+        {
+            writer.Write((int)curve.preWrapMode);
+            writer.Write((int)curve.postWrapMode);
+
+            writer.Write(curve.length);
+            for (int i = 0; i < curve.length; ++i)
+            {
+                var keyframe = curve.keys[i];
+                writer.Write(keyframe.time - startTime);
+                writer.Write(keyframe.value);
+            }
+        }
+
+        /// <summary>
+        /// Deserialize an animation curve from binary data, ignoring tangents.
+        /// </summary>
+        public static void ReadBoolCurve(BinaryReader reader, AnimationCurve curve)
+        {
+            curve.preWrapMode = (WrapMode)reader.ReadInt32();
+            curve.postWrapMode = (WrapMode)reader.ReadInt32();
+
+            int keyframeCount = reader.ReadInt32();
+
+            Keyframe[] keys = new Keyframe[keyframeCount];
+            for (int i = 0; i < keyframeCount; ++i)
+            {
+                keys[i].time = reader.ReadSingle();
+                keys[i].value = reader.ReadSingle();
+                keys[i].inTangent = 0.0f;
+                keys[i].outTangent = 0.0f;
+                keys[i].inWeight = 0.0f;
+                keys[i].outWeight = 1.0e6f;
```

Comment:
Nit: It might be worth having 1.0e6f as a constant (since it's being used elsewhere where binary data is involved). Otherwise, it's a somewhat magic number that would possibly be easy to miss by someone else making a change in the future.

Reply:
True.

Note that this value will not end up in exported data, it's just a way to create quasi-boolean animation curves represented by float values. The weight just needs to be arbitrarily large so that interpolation will effectively return zero or one over the whole interval until the value changes with the next keyframe.